### DISCUSSION TITLE
Send magic link from server programmatically

### DISCUSSION THREAD

**Discussion by fanckush** [supabase/supabase#22294] [Questions] [answered]
In the dashboard in the users list, I am able to manually send a new magic link email to the user.

However I have not found a way to do that using the js client. 

* calling `client.auth.signInWithOtp()` from the server side resets the session
* `client.auth.admin.generateLink()` is half the solution, I get the link but I still need to email it
* `client.auth.resend()` also doesn't work because no email has been sent yet

Supabase has this as a button here, so it must be possible via the client, shouldn't it?

![image](https://github.com/supabase/supabase/assets/6278713/2f5ea3ef-a0bc-4272-9e1c-c4ddeacfb6d0)

**Comment by GaryAustin1** [comment, marked as answer]
https://supabase.com/docs/reference/javascript/auth-admin-inviteuserbyemail

**Reply by fanckush** [reply]
@GaryAustin1 thanks for the link. does `inviteUserByEmail` send a _magic link email_? from my understanding it sends an _invitation email_ which uses a different template

**Reply by GaryAustin1** [reply]
If the user exists the magic link template is used, at least that is what the dashboard does.   I’ve not personally used the call.

**Reply by fanckush** [reply]
alright, i'll give it a try! thanks

**Reply by alecastra** [reply]
This does not work. It returns an error `AuthApiError: A user with this email address has already been registered`

**Comment by hafbau** [comment]
What I did was `separateAdminClient.auth.signInWithOtp()` to avoid server side resetting the session
